plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java

```diff
@@ -91,9 +106,12 @@ public void writeBlob(String blobName, InputStream inputStream, long blobSize) t
         if (blobExists(blobName)) {
             throw new FileAlreadyExistsException("blob [" + blobName + "] already exists, cannot overwrite");
         }
-        try (OutputStream stream = createOutput(blobName)) {
-            Streams.copy(inputStream, stream);
-        }
+
+        final UploadMethod method = (blobSize <= blobStore.bufferSizeInBytes()) ? this::executeSingleUpload : this::executeMultipartUpload;
```

Comment:
I fell like a functional interface here doesn't really buy us anything comparing to a simple if statement with two calls.

Reply:
I agree and pushed https://github.com/elastic/elasticsearch/pull/27280/commits/4421ee1e0c876307a6f5da795894533d05d60859